beacon_node/lighthouse_network/src/discovery/subnet_predicate.rs

```diff
@@ -40,12 +37,20 @@ where
                 .as_ref()
                 .map_or(false, |b| b.get(*s.deref() as usize).unwrap_or(false)),
             Subnet::DataColumn(s) => {
-                let mut subnets = DataColumnSubnetId::compute_custody_subnets::<E>(
+                // If ENR does not contain a valid csc, we fall back to `custody_requirement` here.
```

Comment:
I'm divided on this. Let's consider mainnet post-peerdas activation and there's a widespread ENR encoding bug:
- A (if we error): nodes won't be able to connect to each other, we would notice immediately, but risks liveness as you can't peer up
- B (if we fallback): all nodes will appear as regular nodes (no super-nodes). It is not immediately obvious as the only metric would be a decrease in the count of supernodes (**TODO**: add a metric for this).

The non-discoverability of B seems worrying to me. Due to the current multi-client architecture of Ethereum the liveness risks of A don't appear so severe. Issues in ENR encoding should become apparent quickly in any devnet run.

I would prefer to go for A if I'm not missing something

Reply:
We could add a debug log here - however we can't return an error here without changing discv5 code I think, this predicate function expects a `bool` return and  is used in discv5 predicate query [here](https://github.com/sigp/discv5/blob/682b51e317216b1e2ec3d75b541cdff66d27cf32/src/discv5.rs#L691-L699).

We could return `false` to ignore the peer, but IMO if they could serve column requests they're assigned to custody, they're still useful.

> Issues in ENR encoding should become apparent quickly in any devnet run.

There could still be client changes after devnets / testnets - and adding errors / warnings due to peer issues would be confusing for end users as there's nothing they can do to resolve this.

Reply:
I actually ended up with A, but returning `false` instead of an error, if their ENR is bad we just don't connect to them and find other peers instead. 

> Issues in ENR encoding should become apparent quickly in any devnet run.

I think what you said make sense, if we're discovering and not finding peers, it'd be a bug on our side and should be apparent. If we start ignoring peers with invalid ENRs, it's on the client version that they're using and that's fine.